src/support.cpp

```diff
@@ -783,6 +788,213 @@ void AreaSupport::generateOverhangAreasForMesh(SliceDataStorage& storage, SliceM
+    // values as multiple may be calculated when multiple layers are used for z-smoothing of the offsets.
+    // The average of all offset dists is taken for the used varying offset. To account for this the commutative
+    // offset, and the number of offsets $n$ are stored simultaneously. The final offset used is then commutative
+    // equal to commutative_offset / n.
+    SparsePointGridInclusive<std::pair<size_t, double>> offset_dist_at_point(snap_radius);
+
+    // Collection of the various areas we used to calculate the areas for. This is a combination
+    //  - the support distance (this is the support top distance for overhang areas, and support
+    //    bottom thickness for sloped areas)
+    //  - of the delta z between the current layer and layer below (this can vary between the areas
+    //    when we use multiple layers for z-smoothing)
+    //  - the polygon delta; the xy-distance is calculated separately for overhang and sloped areas.
+    //    here either the slope or overhang area is stored
+    std::vector<std::tuple<double, double, Polygons>> z_distances_layer_deltas;
+
+    constexpr LayerIndex layer_index_offset = 1;
+
+    const LayerIndex layer_idx_below = std::max(layer_idx - layer_index_offset, static_cast<LayerIndex>(0));
+    if (layer_idx_below != layer_idx)
+    {
+        auto layer_below = storage.layers[layer_idx_below].getOutlines()
+                               .offset(-close_dist)
+                               .offset(close_dist)
+                               .smooth(close_dist);
+
+        z_distances_layer_deltas.emplace_back(
+            static_cast<double>(support_distance_top),
+            static_cast<double>(layer_index_offset * layer_thickness),
+            layer_current.difference(layer_below)
+        );
+
+        z_distances_layer_deltas.emplace_back(
+            static_cast<double>(support_distance_bot),
+            static_cast<double>(layer_index_offset * layer_thickness),
+            layer_below.difference(layer_current)
+        );
+    }
+
+    const LayerIndex layer_idx_above = std::min(layer_idx + layer_index_offset, static_cast<LayerIndex>(storage.layers.size() - 1));
+    if (layer_idx_above != layer_idx)
+    {
+        auto layer_above = storage.layers[layer_idx_below].getOutlines()
+                               .offset(-close_dist)
+                               .offset(close_dist)
+                               .smooth(close_dist);
+
+        z_distances_layer_deltas.emplace_back(
+            static_cast<double>(support_distance_bot),
+            static_cast<double>(layer_index_offset * layer_thickness),
+            layer_current.difference(layer_above)
+        );
+
+        z_distances_layer_deltas.emplace_back(
+            static_cast<double>(support_distance_top),
+            static_cast<double>(layer_index_offset * layer_thickness),
+            layer_above.difference(layer_current)
+        );
+    }
+
+    int i = 0;
```

Comment:
```suggestion
    int i = 0;
```

Is this variable here looking pretty, or do we actually use it? I couldn't find it